interfaces/cython/cantera/examples/reactors/ic_engine.py

```diff
@@ -141,16 +141,19 @@ def piston_speed(t):
         injector_mfc.set_mass_flow_rate(0)
 
     # perform time integration, refine time step if necessary
+    solved = False
     for n2 in range(4):
-        if n2 is 4:
-            raise 'Error: Refinement limit reached'
         try:
             sim.advance(t_i)
-        except Exception:
+            solved = True
+            break
+        except ct.CanteraError:
             sim.set_max_time_step(1e-6 * 10. ** -n2)
             n_last_refinement = n1
+        if n2 == 3 and solved == False:
```

Comment:
The `n2 == 3` check is redundant, I'd suggest removing it.
```suggestion
        if solved == False:
```

Reply:
Actually, this should just be
```suggestion
        if not solved:
```